docs/api-reference/components.md

```diff
@@ -0,0 +1,69 @@
+# Components
+
+## RouterProvider
+
+The `RouterProvider` component holds the Yarr router context and is responsible for passing it down to underlying components. This component should be used at a level higher than the page components. It takes the router object as props and passes it down. 
+
+### Usage
+
+1. Create a router object using a routes object and other options
+
+```jsx
+// routes is an array of objects describing which component
+// to load for which url location.
+const router = createBrowserRouter({
+  routes,
+  awaitComponent: true
+});
+``` 
+2. Pass the router object into `RouterProvider` as prop.
+
+```jsx
+ <RouterProvider router={router}>
+   <App/>          
+ </RouterProvider>
+```
+
+### Link
+
+The Link component allows a user to transition to a different url location within the app. 
```

Comment:
We should maybe also mention here that the Link component additionally handles "intent" preloading. A mouseover a link will preload code while a mousedown with preload code and data (as an example).